identity/aziot-identityd/src/lib.rs

```diff
@@ -47,105 +48,83 @@ macro_rules! match_id_type {
+    let (file_changed_tx, mut file_changed_rx) = tokio::sync::mpsc::channel(10);
+    let config_directory_path_copy = config_directory_path.clone();
 
-    {
-        let mut api_ = api.lock().await;
+    // Start file watcher using blocking channel
+    std::thread::spawn(move || {
+        let (file_watcher_tx, file_watcher_rx) = std::sync::mpsc::channel();
 
-        log::info!("Provisioning starting.");
-        let provisioning = api_.provision_device().await?;
-        log::info!("Provisioning complete.");
+        // Create a watcher object, delivering debounced events
+        let mut file_watcher =
+            notify::watcher(file_watcher_tx, std::time::Duration::from_secs(10)).unwrap();
 
-        let device_status = if let aziot_identity_common::ProvisioningStatus::Provisioned(device) =
-            provisioning
-        {
-            let curr_hub_device_info = configext::HubDeviceInfo {
-                hub_name: device.iothub_hostname,
-                device_id: device.device_id,
-            };
-            let device_status = toml::to_string(&curr_hub_device_info)?;
-
-            // Only consider the previous Hub modules if the current and previous Hub devices match.
-            let prev_hub_modules = if prev_settings_path.exists() && prev_device_info_path.exists()
-            {
-                let prev_hub_device_info = configext::HubDeviceInfo::new(&prev_device_info_path)?;
-
-                if prev_hub_device_info == Some(curr_hub_device_info) {
-                    let prev_settings = configext::load_file(&prev_settings_path)?;
-                    let (_, prev_hub_modules, _) =
-                        prepare_authorized_principals(&prev_settings.principal);
-                    prev_hub_modules
-                } else {
-                    std::collections::BTreeSet::default()
-                }
-            } else {
-                std::collections::BTreeSet::default()
-            };
-
-            let () = api_
-                .init_hub_identities(prev_hub_modules, hub_modules)
-                .await?;
-            log::info!("Identity reconciliation with IoT Hub complete.");
+        // Add configuration path to be watched
+        file_watcher
+            .watch(config_directory_path_copy, notify::RecursiveMode::Recursive)
+            .unwrap();
 
-            device_status
-        } else {
-            configext::HubDeviceInfo::unprovisioned()
-        };
+        loop {
+            let _ = file_watcher_rx.recv();
+            let _ = file_changed_tx.blocking_send(());
+        }
+    });
 
-        std::fs::write(prev_device_info_path, device_status)
-            .map_err(error::InternalError::SaveDeviceInfo)?;
-        let () = std::fs::write(prev_settings_path, &settings_serialized)
-            .map_err(error::InternalError::SaveSettings)?;
-    }
+    // Start file change listener that asynchronously updates Identity Service
+    tokio::task::spawn(async move {
+        loop {
```

Comment:
```rust
while let Some(()) = file_changed_rx.recv().await {
    let new_settings = ...
}
```

Reply:
Fixed

Reply:
You've put the `while` inside the `loop`. The `while` was meant to replace the `loop`.

Reply:
Oops. Fixed.